synapseclient/models/wiki.py

```diff
@@ -1 +1,1018 @@
+
+        Arguments:
+            owner_id: The ID of the owner entity.
+            offset: The index of the pagination offset.
+            limit: Limits the size of the page returned.
+            synapse_client: Optionally provide a Synapse client.
+        Returns:
+            A list of WikiHeader objects for the entity.
+        """
+        if not owner_id:
+            raise ValueError("Must provide owner_id to get wiki header tree.")
+        headers = []
+        async for item in get_wiki_header_tree(
+            owner_id=owner_id,
+            offset=offset,
+            limit=limit,
+            synapse_client=synapse_client,
+        ):
+            headers.append(cls().fill_from_dict(item))
+        return headers
+
+
+@dataclass
+@async_to_sync
+class WikiPage(WikiPageSynchronousProtocol):
+    """
+    Represents a [Wiki Page](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/v2/wiki/V2WikiPage.html).
+
+    Attributes:
+        id: The unique identifier for this wiki page.
+        etag: The etag of this object. Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle
+            concurrent updates. Since the E-Tag changes every time an entity is updated it is
+            used to detect when a client's current representation of an entity is out-of-date.
+        title: The title of this page.
+        parent_id: When set, the WikiPage is a sub-page of the indicated parent WikiPage.
+        markdown: The markdown content of the wiki page.
+        attachments: A list of file attachments associated with the wiki page.
+        owner_id: The ID of the owning object (e.g., entity, evaluation, etc.).
+        created_on: The timestamp when this page was created.
+        created_by: The ID of the user that created this page.
+        modified_on: The timestamp when this page was last modified.
+        modified_by: The ID of the user that last modified this page.
+        version_number: The version number of this wiki page.
+        markdown_file_handle_id: The ID of the file handle containing the markdown content.
+        attachment_file_handle_ids: The list of attachment file handle ids of this page.
+    """
+
+    id: Optional[str] = None
+    """The unique identifier for this wiki page."""
+
+    etag: Optional[str] = field(default=None, compare=False)
+    """The etag of this object. Synapse employs an Optimistic Concurrency Control (OCC) scheme to handle concurrent
+    updates. Since the E-Tag changes every time an entity is updated it is used to detect
+    when a client's current representation of an entity is out-of-date."""
+
+    title: Optional[str] = None
+    """The title of this page."""
+
+    parent_id: Optional[str] = None
+    """When set, the WikiPage is a sub-page of the indicated parent WikiPage."""
+
+    markdown: Optional[str] = None
+    """The markdown content of this page."""
+
+    attachments: List[Dict[str, Any]] = field(default_factory=list)
```

Comment:
Instead of returning a Dict here, could we instead return back a WikiAttachment dataclass? If one doesn't exist you'd need to create it

Reply:
Here the attachments just a list of file path. For attachment-related endpoints, only url is given. Do you still think this is needed?

Reply:
Looking at the Rest API Docs for the `V2WikiPage`: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/v2/wiki/V2WikiPage.html

[On these lines](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/e56540eb29e272a551fa1a339997e642936bd53f/synapseclient/models/wiki.py#L397-L413) you are setting:

1. `markdown`
2. `attachments`
3. `owner_id`
4. `wiki_version`

But - These fields don't exist on the Rest API docs. When you call the `fill_from_dict`, these would always be wiped out and replaced with `None` or `[]`, since Synapse would never return these values.

I understand how these fields can be used during the store operations - But, how would a user read for these values?